R/modspec.R

```diff
@@ -865,9 +867,29 @@ autodec_find <- function(code) {
   if(length(code)==0) {
     return(NULL)
   }
-  ans <- regmatches(code, regexpr("[._[:alnum:]]+ *=([^=]|$)", code, perl = TRUE))
-  ans <- unique(sub(" *=.?$", "", ans, perl = TRUE))
-  ans[!grepl(".", ans, fixed = TRUE)]
+  m <- regexpr("[._[:alnum:]]+ *=([^=]|$)", code, perl = TRUE) 
+  ans <- regmatches(code, m)
+  if(!length(ans)) return(character(0))
+  ans <- sub(" *=.?$", "", ans, perl = TRUE)
+  has_dot <- grepl(".", ans, fixed = TRUE)
+  code <- code[m > 0]
+  m <- m[m > 0]
+  pre <- substr(code, start = 0, stop = m-1)
+  pre <- trimws(pre, which = "left")
+  if(all(pre=="")) {
+    return(unique(ans[!has_dot]))  
+  }
+  pre <- strsplit(pre, "[ )(}{\\[\\]]", perl = TRUE)
+  p0 <- sapply(pre, "[", 1L)
+  pre <- lapply(pre, rev)
+  p1 <- sapply(pre, "[", 1L)
+  p2 <- sapply(pre, "[", 2L)
+  drop1 <- p1 %in% c("double", "int", "bool", "const", "static", "unsigned")
+  drop2 <- p2 %in% c("static", "const", "unsigned")
+  drop3 <- grepl("::", p0, fixed = TRUE)
```

Comment:
This is what got me into the PR: I noticed that some variables were getting declared globally as `double` and then locally as `evt::ev`. This isn't affecting any functionality, but it's not correct and those variables show up as getting declared as double if you look.